### PR TITLE
frontend/ci: add workflow checking prettier formatting

### PR DESCRIPTION
**Pull request by draaglom**
see #1366

a few considerations:

1. I've set this up as a separate github workflow. I think lots of small workflows is the better pattern for these kinds of checks: it's quite possible that with N different linters/checks, a bigger PR might fail on multiple of them, which leads to annoying fix-push-check pingpong if you have several checks in sequence in a regular action - and dev time is more valuable than an extra marginal 20s of build time. Yes you can write the action itself to parallelize but that can end up surprisingly complex - plus, with caching, the overhead of parallel workflows can be quite small.

2. To do this practically, you need to pin the exact version of prettier/its plugins, as by definition different versions format ever so slightly differently. Note that I've pinned not quite the latest prettier (3.3.1 not 3.3.3) because empirically that appears to be what people are using (3.3.3 wants to change quite a few files).

This may take a tiny bit of editor config at first, if some folks are using a global prettier version not the project prettier by accident - but most editors detect this by default.

3. As a corollary of (2), you'll now need to run `npm run format` for every prettier/plugin version bump or the build will fail.

Example of a failed build:

image

### CODE REVIEW

File: .github/workflows/prettier.yml [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+          cache-dependency-path: front_end/package-lock.json
+      - name: install deps
+        run: npm ci
+        working-directory: front_end
+      - name: prettier version
+        run: npx prettier -v
+        working-directory: front_end
+      - name: check formatting
+        run: npm run format:check
+        working-directory: front_end
```
Comment (ncarazon) [inline .github/workflows/prettier.yml:38]: Could we also add a step to run linters and typescript checks (`npm run lint`)?

**Comment by hlbmtc** [comment]
@draaglom Added `.github/workflows/prettier.yml` placeholder to `metaculus:main`, so you could test CI runs

File: .github/workflows/prettier.yml (hunk 1 above)
Comment (draaglom) [reply .github/workflows/prettier.yml]: can do - I'd rather follow up in a separate PR though if you don't mind - for the reason of consideration (1) above, and also to stop the PR getting too big (I took a look at what it'd take to run linter on everything and there were a good number of changes to solve the errors when I last checked)

**Comment by draaglom** [comment]
@hlbmtc btw I was able to test CI runs already on `draaglom/metaculus`, all appears to be working AFAICT!

File: .github/workflows/prettier.yml (hunk 1 above)
Comment (ncarazon) [reply .github/workflows/prettier.yml]: Yeah, makes sense to do in a separate PR 👍

**Review by ncarazon — APPROVED** [review]
LGTM!

**Pull request merged**